Improve blend tree contrast/margins

ckaiser opened the pull request:
Improved light theme contrast and readability by adding some margins and adjusting some theme stuff that I think might be leftovers from previous PRs that made adjustments to this.

~Fixed graph nodes using `get_size()` instead of `get_minimum_size()` when they weren't resizable which could result in the nodes retaining their original size after changes. I could not find what exactly caused this regression since 4.2 but this seems like it should be the intended behavior anyhow, if the user can't resize the graph node, it should just adjust to its contents automatically.~

| 4.2 | 4.3 | This PR |
|---|---|---|
| <video src="https://github.com/user-attachments/assets/ea67125b-0a40-477c-b769-e45ea1664422" /> |  <video src="https://github.com/user-attachments/assets/e4768bb3-43f7-4a42-a797-64025574d434" /> | <video src="https://github.com/user-attachments/assets/8f99338f-a407-4844-b518-872fd666524f" />

ckaiser committed:
Fix graph node sizing regression, improve blend tree contrast/margins

akien-mga commented on editor/plugins/visual_shader_editor_plugin.cpp:
```diff
@@ -570,6 +571,10 @@ void VisualShaderGraphPlugin::update_theme() {
 	vs_msdf_fonts_theme->set_font(SceneStringName(font), "Label", label_font);
 	vs_msdf_fonts_theme->set_font(SceneStringName(font), "GraphNodeTitleLabel", label_bold_font);
+	if (!EditorThemeManager::is_dark_theme()) {
+		// Override the color to white for light themes
```
```suggestion
		// Override the color to white for light themes.
```

Geometror reviewed (commented):
In which cases does the regression appear? Could you upload an minimal reproduction project?

ckaiser commented:
> In which cases does the regression appear? Could you upload an minimal reproduction project?

I'll try to make an MRP this weekend to check under what circumstances it happens, the only one that was clear was when opening and closing the resource picker in StateMachine nodes in blend trees.

I'm not sure if it's something specific to this particular node (EditorResourceProperty) interacting with the graph node so I'll do some more digging

ckaiser commented:
I've done some testing, and I don't think the issue comes from the graph node itself, but rather something else that's not allowing the resize to trigger and make the node recalculate its size.

For this particular specific case I've removed the minimum size change and added a fix that restores the original behavior in the blend trees graph, since I haven't seen anything else affected by this I think this is the path with the least friction, if slightly hacky

Geometror commented:
I updated the PR title and description to reflect that. 
But do you still have the minimum size patch around? It would be nice to investigate that.

Geometror reviewed (approved):
I rebased and tested this PR locally - looking good!
Currently, the embedded inspector of nodes seems to be broken (won't even show up with error `Parameter "parent inspector" is null.`), but that's unrelated to this PR (verified).

ckaiser commented:
Rebased, and I can confirm the embedded inspector is broken now, unrelated to the PR

passivestar commented:
This PR fixes a problem that I recently noticed in my theme where progress bars with expanded margins wouldn't even fit within the nodes

![g](https://github.com/user-attachments/assets/f9413838-868d-4021-a71c-86b034f6ee38)

I had to [remove expanded margins](https://github.com/passivestar/godot-minimal-theme/commit/62ac648adb9fde713f3b3298448535080c40053a) because of that, making all of the progress bars in the editor smaller. When this PR is merged I can make progress bars bigger again

akien-mga commented:
Thanks!
